### PR TITLE
Uses the latest locale fix from FluxC

### PR DESCRIPTION
**Pull request by antonis**
**Depends on: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2940**

## Description

This PR fixes broken tests (internal ref p1704756207963939-slack-C5ALGJYEP) in `23.9` due to the change in the `locale` parameter introduced in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2937 by including the fix from https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2938

⚠️ **Note:** Please update the FluxC reference accordingly when the FluxC Pr is merged and a new FluxC release is created

-----

## To Test:

* Verify that CI checks are 🟢 
* Change the device language to a language other than English
* Ensure the blogging prompt is in the selected language.

-----

## Regression Notes

1. Potential unintended areas of impact

    N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    N/A

3. What automated tests I added (or what prevented me from doing so)

    Relied on existing [LoginTests](https://github.com/wordpress-mobile/WordPress-Android/blob/release/23.9/WordPress/src/androidTest/java/org/wordpress/android/e2e/LoginTests.kt) and [BlockEditorTests](https://github.com/wordpress-mobile/WordPress-Android/blob/release/23.9/WordPress/src/androidTest/java/org/wordpress/android/e2e/BlockEditorTests.kt)

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by zwarm — APPROVED** [review]
@antonis - all works as expected

- Verify that CI checks are 🟢
- Change the device language to a language other than English
- Ensure the blogging prompt is in the selected language.

In addition, I ran the failing `e2eLoginWithEmailPassword` test locally and 👍 

**NOTE: The FluxC Version has not been updated yet**

File: build.gradle [hunk 1]
```diff
@@ -24,7 +24,7 @@ ext {
     automatticTracksVersion = '3.3.0'
     gutenbergMobileVersion = 'v1.109.3'
     wordPressAztecVersion = 'v1.9.0'
-    wordPressFluxCVersion = '2.59.2'
+    wordPressFluxCVersion = '2940-26956468d5dc5911a79d28fb510cb652a9cee6ee'
```
Comment (oguzkocer) [inline build.gradle]:
```suggestion
    wordPressFluxCVersion = '2.59.3'
```

Comment (oguzkocer) [reply build.gradle]: @antonis CI just finished publishing `2.59.3`. Do you mind applying this change?

Comment (zwarm) [reply build.gradle]: I can do this @oguzkocer FYI: @antonis

Comment (zwarm) [reply build.gradle]: Looks like someone beat me to it. ❤️

Comment (antonis) [reply build.gradle]: Thank you both 🙇  I Committed the suggestion.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by oguzkocer** [comment]
Thank you both! Let's wait until CI is all green and then I'll proceed with the submission.

**Pull request merged**